src/crypto/transform.py

```diff
@@ -1,39 +1,69 @@
 
-from decouple import config
-from google.cloud.storage import Client as BlobClient
-from google.cloud.bigquery import Client as BigQueryClient
+from google.cloud.storage import Client as CSClient # Cloud Storage Client
+from google.cloud.bigquery import Client as BQClient # BigQuery Client
 from google.cloud.storage.bucket import Bucket
 from google.cloud.storage.blob import Blob
 
+from crypto.utils.helpers import service_credentials, time_period
+from crypto.utils.constants import (
+    BUCKET,
+    CLOUD_STORAGE,
+    BIGQUERY,
+    DAY_DIM,
+    MONTH_DIM,
+    INITIAL_LOAD
+)
 
-class Transform:
 
-    def __init__(self) -> None:
-        self._blob_client = BlobClient()
-        self._bq_client = BigQueryClient()
-        self._blob = f"{datetime.today().strftime('%Y-%m-%d')}.json"
+class Transform:
 
-    def _get_bucket(self) -> Bucket:
-        bucket_name = config("BUCKET", cast=str)
-        bucket = self._blob_client.get_bucket(bucket_name)
-        return bucket
+    def __init__(self, storage_client: CSClient, bq_client: BQClient, bucket_name: str) -> None:
+        self.storage_client = storage_client
+        self.bq_client = bq_client
+        self.bucket_name = bucket_name
 
-    def _get_blob(self) -> Blob:
-        bucket = self._get_bucket()
-        blob = bucket.blob(self._blob)
-        with blob.open("r") as file:
-            crypto_data = json.loads(file.read())
-        return crypto_data
+    def bucket(self) -> Bucket:
+        return self.storage_client.bucket(bucket_name=self.bucket_name)
+    
+    def blob(self, blob_name: str) -> Blob:
+        return self.bucket().blob(blob_name=blob_name)
 
-    def _load_facts(self) -> None:
+    def read_blob(self) -> dict:
+        blob_name = f"{datetime.today().strftime('%Y-%m-%d')}.json"
+        blob = self.blob(blob_name=blob_name)
+        if blob.exists():
+            with blob.open("r") as file:
+                crypto_data = json.loads(file.read())
+            return crypto_data
+        return {}
+    
+    @time_period
+    def day_dim_rows(self) -> list:
+        return list(calendar.day_name)
+    
+    @time_period
+    def month_dim_rows(self) -> list:
+        return list(calendar.month_name)[1:]
+    
+    def _date_dim_row(self) -> list:
         pass
+    
+    def load_table(self, table_id: str, rows: list) -> None:
+        table = self.bq_client.get_table(table=table_id)
+        self.bq_client.insert_rows(table=table_id, rows=rows, selected_fields=table.schema)
+        
 
-    def _load_dimensions(self) -> None:
-        pass
+if __name__ == "__main__":
+    storage_client = CSClient(credentials=service_credentials(CLOUD_STORAGE))
+    bq_client = BQClient(credentials=service_credentials(BIGQUERY))
 
+    transform = Transform(storage_client=storage_client, bq_client=bq_client, bucket_name=BUCKET)
 
-if __name__ == "__main__":
-    transform = Transform()
-    transform._get_blob()
\ No newline at end of file
+    if INITIAL_LOAD:
```

Comment:
double-checked, you used cast to bool so ok, I was not sure if this could result in "False" being true, so all good